test/CrudTest.php

```diff
@@ -90,24 +88,28 @@ public function testIllegalUpdate()
             $exThrown = false;
             try {
                 $this->getLDAP()->update($dn, $entry);
-            } catch (Exception\LdapException $e) {
+            } catch (LdapException $e) {
                 $exThrown = true;
             }
             $this->getLDAP()->delete($dn);
             if (! $exThrown) {
                 $this->fail('no exception thrown while illegally updating entry');
             }
-        } catch (Exception\LdapException $e) {
+        } catch (LdapException $e) {
             $this->fail($e->getMessage());
         }
+
+        // This test "manually" handles which exceptions are expected where.
+        // So it does make any "assert*" calls, and does not set any expected exception.
+        // Because of this, phpunit will flag this as a risky test,
+        // so assert something before finishing the test.
+        $this->assertTrue(true);
```

Comment:
Also here: this will fail at static analysis (coming up in future).

Anything valuable we can assert upon at all?

Reply:
I will be offline for a couple of hours. I will look at these tonight. They are all only in test code and Travis script so do not effect "real release". If you want to move this along and get a "real release" then feel free to merge this and I can sort this stuff out in another PR. Or wait...

(This PR seems to be dragging on "forever" - each time I address review comments, then more come from the next review)

Reply:
> (This PR seems to be dragging on "forever" - each time I address review comments, then more come from the next review)

That's because as the diff becomes clearer, so do the issues with it.

It's always push and pull though: you have to consider that once this is "done", it is work on the maintainers to chase any problems with it long-term, so we do indeed pay a lot of attention to detail too.

Reply:
I changed it to check the value of `getLastErrorCode()`